### PR TITLE
add provider named extension in dicehub

### PR DESCRIPTION
**Pull request by ZLValien**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:

add provider named extention in dicehub

### CODE REVIEW

**Comment by sfwn** [comment]
ci failed.

File: pkg/common/apis/header.go [hunk 1]
```diff
@@ -56,6 +56,11 @@ func HTTPLanguage(r *http.Request) i18n.LanguageCodes {
 	return langs
 }
 
+// GetLang .
+func GetLang(ctx context.Context) string {
+	return GetHeader(ctx, "Lang")
```
Comment (sfwn) [inline pkg/common/apis/header.go:61]: lower case in grpc.

**Review by sfwn — APPROVED** [review]

File: pkg/common/apis/header.go (hunk 1 above)
Comment (ZLValien) [reply pkg/common/apis/header.go]: done

File: modules/dicehub/extension/db/extension.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+			CreatedAt: timestamppb.New(ext.CreatedAt),
+			UpdatedAt: timestamppb.New(ext.UpdatedAt),
+			IsDefault: ext.IsDefault,
+			Public:    ext.Public,
+		}
+
+	} else {
+		diceData, _ := yaml.YAMLToJSON([]byte(ext.Dice))
+		specData, _ := yaml.YAMLToJSON([]byte(ext.Spec))
+		swaggerData, _ := yaml.YAMLToJSON([]byte(ext.Swagger))
```
Comment (Effet) [inline modules/dicehub/extension/db/extension.go:52]: error should be checked

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dicehub/extension/db/extension.go (hunk 2 above)
Comment (ZLValien) [reply modules/dicehub/extension/db/extension.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**